patches/rust-libc/0001-managarm-initial-port.patch

```diff
@@ -0,0 +1,584 @@
+From 9de522a27f50ee54f4e8ebb8dd689323ffa0706e Mon Sep 17 00:00:00 2001
+From: Matt Taylor <mstaveleytaylor@gmail.com>
+Date: Thu, 8 Apr 2021 22:08:55 +0100
+Subject: [PATCH] managarm: initial port
+
+---
+ src/unix/mlibc/mod.rs | 550 ++++++++++++++++++++++++++++++++++++++++++
+ src/unix/mod.rs       |   3 +
+ 2 files changed, 553 insertions(+)
+ create mode 100644 src/unix/mlibc/mod.rs
+
+diff --git a/src/unix/mlibc/mod.rs b/src/unix/mlibc/mod.rs
+new file mode 100644
+index 000000000..409ac74f5
+--- /dev/null
++++ b/src/unix/mlibc/mod.rs
+@@ -0,0 +1,550 @@
++// TODO: Should padding be public?
++
++// Basic data types
++pub type c_char = i8;
++pub type c_long = i64; // TODO ?
++pub type c_ulong = u64; // TODO ?
++pub type wchar_t = ::c_int; // TODO ?
```

Comment:
What are these TODOs?

Reply:
Ah yes, I forgot about these. I wasn’t entirely sure what type `wchar` should be, and in future we’ll need to change the definition of long based on the target pointer size. But as we only support x86_64 for now it should be fine